### ISSUE TITLE
Performing a TLS handshake is very slow

### ISSUE THREAD

**Issue by renkun-ken** [getlantern/lantern#5751]
I'm using the latest version of lantern (v3.7.3) on Ubuntu 16.04.2. From recent versions, I notice that each time I visit a website in Firefox, the status label says "Performing a TLS handshake to <website>" and it may take more than 10 seconds before downloading the contents. Quit or disable lantern then the problem is gone. 

I'm wondering what makes it so slow doing this?

**Comment by mikedchavez1010** [comment]
Firefox defaults to using the system proxy, which is managed by lantern by default. Please change Firefox's proxy setting to No proxy in Preferences -> Advanced -> Network -> Settings if you're not running Firefox to use lantern to work alongside.

**Comment by gbutler69** [comment]
Yikes! I was about to give up on Firefox due to how slow it has gotten. When did this change where Firefox defaults to using the System Proxy? It made it almost unusable and kept getting worse with each of the last few releases of FF. I thought it was supposed to be getting faster as we move towards 57, but, it was getting slower to the point of unusability. I finally downloaded Vivaldi the other day hoping to find something that had the features I liked from FF but wasn't dog slow. This is terrible! Having it default to a SLOW SLOW SLOW proxy like this silently almost seems like an attempt at undermining FF. I just switched the "Use System Proxy" off in FF settings and now it works great! This is a HUGE problem for FF. Someone needs to get the word out about this!
